### DISCUSSION TITLE
Rename de-CH to gsw?

### DISCUSSION THREAD

**Discussion by dontinelli** [OHF-Voice/intents#2804] [_General]
In course of playing around with Voice PE I noticed an inconsistency in the naming of the Swiss German language. 

Background is that currently there is not yet a reliable and ready-to-use STT engine for Swiss German (local dialect) available. Azure provides a model that translates Swiss German into Swiss Standard German. However, the syntax and also the vocabulary are not identical to the general high German ("de" or "de-de"). The Swiss Standard German is used for writing (newspaper, official publications) or formal speech (e.g. in parliament) in Switzerland. This leads to a situation where we basically have three related but still different languages in this context, (high) German, Swiss Standard German and Swiss German.

When starting with creating intents for Swiss German, the language has been setup as "de-CH". The intents are in local dialect, as it is this language that is used orally in informal context (and therefore would be the language of choice for a voice assistant). it seems, though, that "de-CH" is not the correct classification, "gsw" would be more appropriate. It could therefore make sense to rename the language to "gsw". I'm not yet sure if it is worthwhile to add "de-CH" in addition to "gsw". As written above, the Swiss Standard German is rather used in formal context, so I would not expect that a lot of users would use it natively. But it could be used to enable STT with the currently existing models.

I'm curious what arguments and thoughts you have on this topic.

**Comment by dontinelli** [comment]
Doesn't seem to reach a lot of attention... Nevertheless it would be helpful to get some inputs and in the end a decision against or in favour. Any thoughts, @synesthesiam ?

I just noticed than in Lokalize Swiss German is also referenced to as "gsw":
![grafik](https://github.com/user-attachments/assets/c1f04280-dc50-4b6d-8546-fe1dac93c0f1)

**Comment by dontinelli** [comment]
@synesthesiam: Any thoughts on this? I saw that quite a lot of work is coming up with the transition of intents repo to slot combinations and I would rather not put too much effort into de-CH with the future unclear.

**Comment by synesthesiam** [comment]
The language names are meant to match what's in Home Assistant, so gsw wouldn't work well even if it's technically correct.

Would Swiss German speakers actually want to speak the dialect that Azure supports to their voice assistants? It sounds like what's needed is a Whisper model that's trained on the gsw dialect.

**Reply by dontinelli** [reply]
Yes, one thing would be a gsw STT (Azure or whatever), but then of course the second step would be that this is correctly handled in HA. If the dialect is de_CH instead of gsw, we have a mismatch.

**Reply by dontinelli** [reply]
But your answer points to an important problem, that also the naming in HA is wrong.
I don't think that there are sufficient differences between de and de-CH that a High Swiss German translation would be needed. Currently de-CH in HA is dialect and hence also should be gsw.

**Reply by synesthesiam** [reply]
@c0ffeeca7 What do you think?

**Reply by c0ffeeca7** [reply]
not sure I can bring value to this conversation. 
- the current content in Home Assistant sounds to me like a Bernese variant, which is a dialect I mostly understand but don't speak. So I don't feel I can contribute.
  - If I understand correctly, gsw is an ISO 639-3 code covering all Alemannic dialects, [even including Alsacian](https://en.wikipedia.org/wiki/ISO_639:gsw) so it should cover the Swiss-German dialects
  - I think de-CH was chosen as a language code because elsewhere in the repo we use the language plus locale code (zh-hk, pt-BR), not the three-letter code.
